### PR TITLE
Product Page - Product Images/Details

### PR DESCRIPTION
**Pull request by MehmetSunguralp**
image
image

Renders product images and details up to product ID

### CODE REVIEW

File: src/app/product/[productId]/page.tsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { ProductDetail } from "@/components/ProductDetail/ProductDetail";
+
+export default function ProductPage({ params }: { params: { productId: string } }) {
+	const { productId } = params;
+	return (
+		<>
```
Comment (kumbu132) [inline src/app/product/[productId]/page.tsx:6]: When you only have one child, its not necessary to have the `<> </>` wrapper

File: src/components/ProductDetail/ProductDetail.tsx [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+	// Fetch products and find the matching product by productId
+	useEffect(() => {
+		fetchProducts().then((data) => {
+			const foundProduct = data.products.find((p: { id: string }) => p.id == productId);
+			setProduct(foundProduct || null);
+			if (foundProduct) {
+				setMainImage(foundProduct.images[0]);
+			}
+		});
+	}, [productId]);
```
Comment (kumbu132) [inline src/components/ProductDetail/ProductDetail.tsx:41]: You can opt to use an sync useCallback that is called from the useEffect here.

File: src/components/ProductDetail/ProductDetail.module.scss
Comment (kumbu132) [inline src/components/ProductDetail/ProductDetail.module.scss:1]: Mostly okay. The biggest potential issue is directly styling HTML elements. Try to get in the habit of styling classes

File: src/components/ProductCard/ProductCard.tsx [hunk 3]
```diff
@@ -27,10 +27,11 @@ export const ProductCard: React.FC<ProductCardProps> = ({ id, thumbnail, title,
 	//Calculate discounted price
 	const discount = Math.floor(discountPercentage);
 	const newPrice = discount !== 0 ? price - (price * discount) / 100 : price;
+
 	return (
-		<Link href={""} className={styles.productCard} title={title} id={String(id)}>
+		<Link href={"product/" + id} className={styles.productCard} title={title} id={String(id)} prefetch={true}>
 			<div className={styles.thumbnailWrapper}>
-				<Image src={thumbnail} alt="asd" fill sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw" />
+				<Image src={thumbnail} alt=" asd" fill sizes="(max-width: 768px) 100vw, (max-width: 1200px) 50vw, 33vw" />
```
Comment (kumbu132) [inline src/components/ProductCard/ProductCard.tsx:34]: You should give a more meaningful name for the alt text

**Review by kumbu132 — CHANGES_REQUESTED** [review]

File: src/app/product/[productId]/page.tsx (hunk 1 above)
Comment (MehmetSunguralp) [reply src/app/product/[productId]/page.tsx:6]: I'll have more components here like review section of the specific product

File: src/components/ProductDetail/ProductDetail.module.scss [hunk 4]
```diff
@@ -130,13 +130,13 @@
...
-			img {
+			.starIcon {
 				width: 20px;
 				height: 20px;
 			}
 			.rating {
-				span {
-					color: $gray;
+				.rate {
+					color: $gray; /*  */
```
Comment (kumbu132) [inline src/components/ProductDetail/ProductDetail.module.scss:139]: you have added `/* */`. why?

Comment (MehmetSunguralp) [reply src/components/ProductDetail/ProductDetail.module.scss:139]: I belive it was a typo

**Review by kumbu132 — APPROVED** [review]

**Pull request merged**